subprojects/core/src/main/java/org/gradle/groovy/scripts/internal/GroovyScriptClassCompiler.java

```diff
@@ -110,16 +114,16 @@ public <T extends Script, M> CompiledScript<T, M> compile(
         HashCode sourceHashCode = source.getResource().getContentHash();
         RemappingScriptSource remapped = new RemappingScriptSource(source);
         ClassLoader classLoader = targetScope.getExportClassLoader();
-        File outputDir = doCompile(target, templateId, sourceHashCode, remapped, classLoader, operation, verifier, scriptBaseClass);
+        GroovyScriptCompilationOutput output = doCompile(target, templateId, sourceHashCode, remapped, classLoader, operation, verifier, scriptBaseClass);
 
-        File genericClassesDir = classesDir(outputDir, operation);
-        File metadataDir = metadataDir(outputDir);
-        // TODO: Move instrumentation to the execution engine and remove the remapping or move remapping to the non-cacheable unit of work
-        ClassPath remappedClasses = remapClasses(genericClassesDir, remapped);
+        File instrumentedJar = output.getInstrumentedJar();
+        File metadataDir = output.getMetadataDir();
+        // TODO: Remove the remapping or move remapping to the non-cacheable unit of work?
```

Comment:
❓ What's "the non-cacheable work"? Is this something we already have, or something to be potentially created later?

Reply:
Hehe, it's just means that we should not cache that unit of work. Changed `the non-cacheable unit of work` to `an uncached unit of work`. Does it make more sense now?